spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/MongoOperations.java

```diff
@@ -629,6 +631,36 @@
 	<T> T findById(Object id, Class<T> entityClass, String collectionName);
 
 	/**
+	 * Map the results of a Query-by-Example query onto the given {@code probe} class. The collection is determined from the {@code probe} type.
+	 * <p/>
+	 * The objects are converted from the MongoDB native representation using an instance of {@see MongoConverter}. Unless
+	 * configured otherwise, an instance of MappingMongoConverter will be used.
+	 * <p/>
+	 * The query is constructed from the {@code probe}.
+	 *
+	 * @param probe the probe to be used for the Example, must not be {@literal null}.
+	 * @param <S>
+	 * @param <T>
+	 * @return the List of converted objects
+	 */
+	<S extends T, T> List<T> findByExample(S probe);
```

Comment:
I get compile errors here due to `S` referring to `T` defined later, inverting the declaration should fix that. Same below.